### PR TITLE
Add longoptions support for psbasemap.

### PR DESCRIPTION
**Pull request by rbdavis**
**Description of proposed changes**

Python differences:

  No differences.

Julia differences:

  No differences.

Fixes #

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: src/longopt/psbasemap_inc.h [hunk 1]
```diff
@@ -25,26 +25,20 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
 		  short_directives,    long_directives,
 		  short_modifiers,     long_modifiers,
 		  transproc_mask */
-	{ 0, 'A', "",
-	          "",                  "",
-	          "",                  "",
+	{ 0, 'A', "polygon",           "", "", "", "", GMT_TP_STANDARD },
+	{ 0, 'D', "inset|inset_box",
+	          "g,j,J,n,x",         "mapcoords,inside,outside,boxcoords,plotcoords",
+	          "r,s,t,u,w,j,o",     "corners,outfile,translate,units,width,janchor,offset",
```
Comment (joa-quim) [inline src/longopt/psbasemap_inc.h:31]: I perfectly understand the difficulty with this one `janchor` (the difference of the two _anchors_ in the option are difficult to grasp), but carrying the letter modifier (**j**) into the long option name is not much helpfull either.

Comment (PaulWessel) [reply src/longopt/psbasemap_inc.h:31]: This is about upper and lower case j|J, right?  How about inanchor and outanchor?

**Review by PaulWessel — APPROVED** [review]

**Comment by joa-quim** [comment]
`janchor` is not good. If we have in other modules it should be replaced there too.

**Comment by PaulWessel** [comment]
Can't it just be justify|justification instead?

**Comment by joa-quim** [comment]
No because we have also **+j** modifiers that take that word, but `inanchor | outanchor` are a good ones.
The point of `janchor` is that it's carrying the one letter code into the long option name, which defeats the all purpose of long options.

**Pull request merged**